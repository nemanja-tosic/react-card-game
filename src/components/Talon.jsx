// @flow
import React from 'react';

import { Card } from '../models/card';
import CardComponent from './Card';

import './Talon.css';

const Talon = ({cards, cardRef}: { cards: Array<Card>, cardRef: any }) => (
  <div className="Talon">
    {cards.map((card, index) => (
      <div key={index} className="Talon__card">
        <CardComponent card={card} cardRef={cardRef}/>
      </div>
    ))}
  </div>
);

export default Talon;
