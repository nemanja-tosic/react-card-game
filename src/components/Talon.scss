@import './variables';

.Talon {
  position: relative;
  display: flex;
  align-items: center;

  &__card {
    width: 100%;
    position: absolute;
  }

  &__card:nth-child(1) {
    margin-top: 40%;
  }

  &__card:nth-child(2) {
    margin-left: -40%;
  }

  &__card:nth-child(3) {
    margin-top: -40%;
  }

  &__card:nth-child(4) {
    margin-left: 40%;
  }
}
