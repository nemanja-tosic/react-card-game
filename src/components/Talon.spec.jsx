// @flow
import React from 'react';
import { shallow } from 'enzyme';

import { TestCards } from '../test/support/test-cards';
import Talon from './Talon';
import CardComponent from './Card';

describe('Talon', () => {
  let component;
  let cardRefSpy;

  const cards = [TestCards.create()];

  beforeEach(() => {
    cardRefSpy = jest.fn();

    component = shallow(<Talon cards={cards} cardRef={cardRefSpy}/>)
  });

  it('should display all the provided cards', () => {
    expect(component.getNode()).toMatchSnapshot();
  });

  it('should call the cardRef prop when a card ref is updated', () => {
    const cardComponent = component.find(CardComponent).at(0);
    const tracer = {};

    cardComponent.prop('cardRef')(tracer);

    expect(cardRefSpy).toHaveBeenCalledWith(tracer)
  });
});
