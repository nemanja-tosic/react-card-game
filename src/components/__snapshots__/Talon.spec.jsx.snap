// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Talon should display all the provided cards 1`] = `
<div
  className="Talon"
>
  <div
    className="Talon__card"
  >
    <CardComponent
      card={
        Card {
          "code": "JS",
          "imageUrl": "https://deckofcardsapi.com/static/img/JS.png",
        }
      }
      cardRef={[Function]}
    />
  </div>
</div>
`;
